### DISCUSSION TITLE
Adding a strange device

### DISCUSSION THREAD

**Discussion by moskovskiy82** [Koenkk/zigbee2mqtt#14621] [Devices]
Got a small RGBWW controller. States Zigbee3. Read through the docs for adding a new device and have some questions left.
I believe it's a https://www.zigbee2mqtt.io/devices/GD-ZCRGB012.html as the model number is the same and they look quite the same. But upon pairing get

Debug 2022-10-23 22:40:32Received Zigbee message from '0x804b50fffe092251', type 'raw', cluster '65518', data '{"data":[25,3,129],"type":"Buffer"}' from endpoint 1 with groupID 0
Warning 2022-10-23 22:40:32Received message from unsupported device with Zigbee model 'A0F' and manufacturer name 'Light'
Warning 2022-10-23 22:40:32Please see: https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html
Info 2022-10-23 22:42:01Device '0x804b50fffe092251' joined
Info 2022-10-23 22:42:01MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251"},"type":"device_joined"}'
Info 2022-10-23 22:42:01MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"0x804b50fffe092251"},"type":"device_connected"}'
Info 2022-10-23 22:42:01Starting interview of '0x804b50fffe092251'
Info 2022-10-23 22:42:01MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251","status":"started"},"type":"device_interview"}'
Info 2022-10-23 22:42:01MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"0x804b50fffe092251"},"type":"pairing"}'
Debug 2022-10-23 22:42:01Device '0x804b50fffe092251' announced itself
Info 2022-10-23 22:42:01MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251"},"type":"device_announce"}'
Info 2022-10-23 22:42:01MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x804b50fffe092251"},"type":"device_announced"}'
Info 2022-10-23 22:42:02Successfully interviewed '0x804b50fffe092251', device has successfully been paired
Warning 2022-10-23 22:42:02Device '0x804b50fffe092251' with Zigbee model 'A0F' and manufacturer name 'Light' is NOT supported, please follow https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html
Info 2022-10-23 22:42:02MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":null,"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251","status":"successful","supported":false},"type":"device_interview"}'
Info 2022-10-23 22:42:02MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"friendly_name":"0x804b50fffe092251","supported":false},"type":"pairing"}'
Debug 2022-10-23 22:42:10Device '0x804b50fffe092251' announced itself
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251"},"type":"device_announce"}'
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x804b50fffe092251"},"type":"device_announced"}'
Debug 2022-10-23 22:42:11Device '0x804b50fffe092251' announced itself
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251"},"type":"device_announce"}'
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x804b50fffe092251"},"type":"device_announced"}'
Debug 2022-10-23 22:42:11Device '0x804b50fffe092251' announced itself
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251"},"type":"device_announce"}'
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x804b50fffe092251"},"type":"device_announced"}'
Debug 2022-10-23 22:42:11Device '0x804b50fffe092251' announced itself
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x804b50fffe092251","ieee_address":"0x804b50fffe092251"},"type":"device_announce"}'
Info 2022-10-23 22:42:11MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x804b50fffe092251"},"type":"device_announced"}'

So the questions

1. Can i just name the js. definition file anyway i want? Like ZCRGB012.js?
2. Giderwell.js looks quite different from examples

const extend = require('../lib/extend');

module.exports = [
    {
        zigbeeModel: ['A10'],
        model: 'GD-ZCRGB012',
        vendor: 'GIDERWEL',
        description: 'Smart Zigbee RGB LED strip controller',
        extend: extend.light_onoff_brightness_color({supportsHS: false}),
    },
];

Do i need all the header stuff? Or just stick to giderwell and change zigbee model, vendor and model?

**Comment by moskovskiy82** [comment]
For question 2 definetly not.

:/usr/share/hassio/share/zigbee2mqtt$ cat gbrdwc12.js
const extend = require('../lib/extend');
\Gives an error 
![image](https://user-images.githubusercontent.com/15857592/197419601-86ac9515-6123-4c0c-82ef-0ffe42905191.png)

**Reply by Danit2** [reply]
You need the header stuff.

You can try this external converter:
```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');

module.exports = [
    {
        zigbeeModel: ['A10'],
        model: 'GD-ZCRGB012',
        vendor: 'GIDERWEL',
        description: 'Smart Zigbee RGB LED strip controller',
        extend: extend.light_onoff_brightness_color({supportsHS: false}),
    },
];
```

**Reply by moskovskiy82** [reply]
> You need the header stuff.
> 
> You can try this external converter:
> 
> ```
> const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
> ```

Thanks it worked. Almost. On and Off is fine. Channels are mixed up. 
red is blue
green is red
blue is red
And the white channel doesn't show up

![image](https://user-images.githubusercontent.com/15857592/197620560-abdfaccf-826d-40e9-9bad-0f7f3641443e.png)

Would appreciate where to go next.

Also a question how to rename the entity and the model in z2m so if it will be successfull 

While changing colors and brightness - all i get in the logs is 

```
Debug 2022-10-24 23:30:49Received MQTT message on 'zigbee2mqtt/0x804b50fffe092251/set' with data '{"state":"ON"}'
Debug 2022-10-24 23:30:49Publishing 'set' 'state' to '0x804b50fffe092251'
Info 2022-10-24 23:30:49MQTT publish: topic 'zigbee2mqtt/0x804b50fffe092251', payload '{"brightness":254,"color":{"x":0.14912280701754385,"y":0.06140350877192982},"color_mode":"xy","linkquality":54,"state":"ON"}'
Debug 2022-10-24 23:30:56Received MQTT message on 'zigbee2mqtt/0x804b50fffe092251/set' with data '{"color":{"x":0.49107142857142855,"y":0.42857142857142855}}'
Debug 2022-10-24 23:30:56Skipping state because of Home Assistant
Debug 2022-10-24 23:30:56Publishing 'set' 'color' to '0x804b50fffe092251'
Debug 2022-10-24 23:30:57Missing colorTempPhysicalMin and/or colorTempPhysicalMax for endpoint 0x804b50fffe092251!
Info 2022-10-24 23:30:57MQTT publish: topic 'zigbee2mqtt/0x804b50fffe092251', payload '{"brightness":254,"color":{"x":0.49107142857142855,"y":0.42857142857142855},"color_mode":"xy","linkquality":57,"state":"ON"}'
```

**Reply by Danit2** [reply]
Ahh, sorry it is a RGBWW. 

Then you have to change the extend part to: 

```
light_onoff_brightness_colortemp_color()
```

When is not working, plead send a screenshot of the cluster tab.

**Comment by moskovskiy82** [comment]
Worked like a charm. Almost. It's a RGBW controller. Do i undesrstand correctly thet then extend part in the brackets should look like 

({colorTempRange: [153, 555]}), ?




And another question is how to give this a proper name. As currently it's producer Light
